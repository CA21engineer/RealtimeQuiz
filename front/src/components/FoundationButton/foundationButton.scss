.foundation__button {
  min-width: 220px;
  height: 64px;
  border-radius: 32px;
  font-size: 1.2rem;
  background-color: rgba(0,0,0,0);
  border: #FFFFFF 2px solid;
  color: #FFFFFF;
  cursor: pointer;

  &:disabled {
    opacity: 0.3;
    cursor: auto;
  }

  &:hover {
    border: #FFFFFF 3px solid;
  }

  &:active {
    background-color: rgba(0, 0, 0, 0.18);
  }
}
