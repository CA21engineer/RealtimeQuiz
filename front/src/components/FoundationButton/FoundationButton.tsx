import React from 'react';
import './foundationButton.scss';

type Props = {
  label: string;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
};

export const FoundationButton: React.FC<Props> = ({ label, onClick }) => {
  return (
    <button type="button" className="foundation__button" onClick={onClick}>
      {label}
    </button>
  );
};
