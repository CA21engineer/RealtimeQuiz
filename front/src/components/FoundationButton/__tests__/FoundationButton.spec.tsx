import React from 'react';
import renderer from 'react-test-renderer';
import { FoundationButton } from '../FoundationButton';

test('FoundationButton', () => {
  const component = renderer.create(
    <FoundationButton label="button label" onClick={() => console.log} />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
