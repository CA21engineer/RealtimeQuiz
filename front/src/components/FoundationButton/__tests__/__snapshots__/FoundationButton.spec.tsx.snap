// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoundationButton 1`] = `
<button
  className="foundation__button"
  onClick={[Function]}
  type="button"
>
  button label
</button>
`;
